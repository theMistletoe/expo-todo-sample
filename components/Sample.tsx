import React, { useState } from 'react';
import { Button, Text, View } from 'react-native';

type Props = {
  text: string
}

const Sample = (props: Props) => {
  const [number, setNumber] = useState<number>(0);

  const handlePress = () => {
    const updateNumber = number + 1;
    setNumber(updateNumber);
  };

  return (
    <View>
      <Text>
        {props.text}!!!
      </Text>
      <Text>{number}</Text>
      <Button onPress={handlePress} title="increment" />
    </View>
  );
};

export default Sample;