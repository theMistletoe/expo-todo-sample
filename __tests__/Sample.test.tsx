import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import Sample from '../components/Sample';

describe("test Sample", () => {
  it('renders correctly with defaults', () => {
    const comp = renderer.create(<Sample text="XxXxX" />)
    expect(comp).toMatchSnapshot()
  })

  it('increments number by pressed button', () => {
    const { getByText } = render(<Sample text="XxXxX" />);
    fireEvent.press(getByText("increment"));
    fireEvent.press(getByText("increment"));
    expect(getByText("2")).toBeTruthy();
  })
})