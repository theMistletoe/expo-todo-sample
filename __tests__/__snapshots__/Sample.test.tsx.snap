// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Sample renders correctly with defaults 1`] = `
<View>
  <Text>
    XxXxX
    !!!
  </Text>
  <Text>
    0
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        increment
      </Text>
    </View>
  </View>
</View>
`;
